59.md

```diff
@@ -0,0 +1,55 @@
+NIP-59
+======
+
+Gift Wrap
+-------------------------------
+
+`draft` `optional` `author:kieran`
+
+This NIP defines a simple method to protect event metadata leakage by using randomly generated keys.
+
+A gift wrap event is a `kind 1059` event which wraps another event with a single use key.
+
+This event kind can be used to wrap DM's or other events which are considered private.
+
+This is similar in concept to Onion Routing.
+
+To construct a wrapped `kind 4` you first start by producing the `kind 4` event as normal, then create a random key and produce the `kind 1059` by encrypting the `kind 4` as a JSON string and storing it in `content` of the `kind 1059` using the same `p` tag to notify the recipient.
+
+Example wrapped event:
+```json
+{
+    "id": "3731dc639ada388aa5ee8c3404351868d4dd0fdcac00183e540669ddc25e8ad4",
+    "pubkey": "4d02c0b4e16db38b80fa3b46548417a4ccc975337e29bc22da72132cf69f3b4a",
+    "content": "7sW2yHvC5o/x6b0xCKPpH8KZJQeQYxt3Y53mxMV53yGQ5kn/jgMM74k+vfwegYBY9zjlQeTvyF5gYuJIHM4KN2SZq7zaCIKqTqMjYhLPI4VjIvowhmMuuZgRlB7cestrYh6OSSteICye6sxkS1NdpON31u+JNK/GHFVRJqfYF4Fnzaka2u9WHQKmm9/nN040OHZ41eT3YUWpaZHupVx1fuN6UoeyPjb2PW4yiwxD72LhlAoNGjR+otpWZ3kD6LU3Btrz8btjSD272wJXT/FBX5Gs59BMCDeJ4eReNfbbVBB5mlXvVqVC/tnHntVgNvAP5GsZE56/4t+kLqql0dqZU/haC8wpOxqtzqzhzxTHxS/IQ3HYpDnU8XuXUqNw1zOQhp0F65rXZUf9nN0gIa/zGa35fK2ek2p8K/3exKssO7fVxaMeTVxP7jEoCmn47gsy5mv4fH7MtxoWvWMapvGFLZf58NPy54ktvldOdGp9BJLO+6/2P7KJqb7bvu0EIPdYCcUt+muYwiJCUoHNFfFQwKrrpr23XuPQlPYNPUs6orbNBs/Oj8vMfZC5hb/AUaNanGtsqsBgqfEufyonBtihMzLZqn9OZNWHE7lYfnhSOZnUjStl03Bd94Qx1rp7IHZN?iv=cIiZuLb2FMsLzc+yYf+3IQ==",
+    "kind": 1059,
+    "created_at": 1682286690,
```

Comment:
`created_at` can be made optional to save space, verifier assumes `created_at` is the same, and verifies the sig as if it is present (not as if it is excluded!).